### PR TITLE
Unit Tests for TabLabelProvider (JENKINS-56721)

### PR DESCRIPTION
**Pull request by RedTo**
Added unit tests for TabLabelProvider#getPackageName.
And also some tests for TabLabelProvider#getPackages.

### CODE REVIEW

**Comment by RedTo** [comment]
I'm sorry, but unfortunately I'm not able to add the <kbd>[work-in-progress](https://github.com/jenkinsci/warnings-ng-plugin/labels/work-in-progress)</kbd>label...

File: src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package io.jenkins.plugins.analysis.core.model;
+
+import org.junit.jupiter.api.Test;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.Report;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java]: Please use AssertJ. That is checked by an architecture test: https://travis-ci.org/jenkinsci/warnings-ng-plugin/builds/514856410#L6587

File: src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        report.add(issue);
+
+        return new TabLabelProvider(report);
+    }
+
+    @Test
+    void shouldReturnPackageNameJava() {
+        TabLabelProvider tabLabelProvider = createTabLabelProvider("Testfile.java");
+
+        assertEquals(tabLabelProvider.getPackageName(), Messages.Tab_Package());
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java]: `assertThat(tabLabelProvider).hasPackageName(...)`

**Comment by RedTo** [comment]
Hmm... okay, I see, the test class should not be public.

**Comment by RedTo** [comment]
I implemented my own version of Assertions, SoftAssertions and also one for TabLabelProviderAssert. But I'm not quite sure if this was the right way, I just saw this in the IssueTests.

File: src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java (hunk 1 above)
Comment (RedTo) [reply src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java]: I changed the assertion library to assertj.

File: src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java (hunk 2 above)
Comment (RedTo) [reply src/test/java/io/jenkins/plugins/analysis/core/model/TabLabelProviderTest.java]: I implemented Assertions, SoftAssertions and TabLabelProviderAssert with assertThat and hasPackageName + hasPackages

**Comment by uhafner** [comment]
The maven Assertion generator had some bugs when we wrote an own Assertion class for {{Issue}}. I should remove these from the project. I see no special feature that is *not* provided by the generated Assertion class `io.jenkins.plugins.analysis.core.model.TabLabelProviderAssert` now. It is referenced from `io.jenkins.plugins.analysis.core.assertions.Assertions`. 

So please remove that part, I would like to keep the code base as small as possible.

**Comment by RedTo** [comment]
I tried to use the generated one, but then I got an architecture error because the test class has to be public to be accessible from outside. But maybe I did something wrong. I will have a closer look.

**Comment by uhafner** [comment]
The test class should be package private indeed. But you are accessing the Assertions from the class and not the other way round. So it basically should work.

**Comment by RedTo** [comment]
I have deleted all the unneeded files and now it should be like it has to be :-)
Sorry for the inconveniences...

**Pull request merged**